cmd/common/snapshot/process.go

```diff
@@ -35,6 +40,91 @@ type ProcessParser struct {
 	outputConfig *commonutils.OutputConfig
 }
 
+type nodeProcess struct {
+	event    *types.Event
+	children []*nodeProcess
+}
+
+func createTree(processes []*types.Event) (*nodeProcess, error) {
+	var root *nodeProcess
+
+	nodes := make(map[int]*nodeProcess, 0)
+	// Create a node for each processes.
+	for _, process := range processes {
+		nodes[process.Pid] = &nodeProcess{
+			event:    process,
+			children: make([]*nodeProcess, 0),
+		}
+	}
+
+	// Link all nodes together.
+	for _, node := range nodes {
+		ppid := node.event.ParentPid
+		if _, ok := nodes[ppid]; !ok {
+			root = node
+
+			continue
+		}
+
+		nodes[ppid].children = append(nodes[ppid].children, node)
+	}
+
+	if root == nil {
+		return nil, fmt.Errorf("container has no root process")
+	}
+
+	return root, nil
+}
+
+func treeToString(node *nodeProcess, front string) string {
+	line := fmt.Sprintf("|-%s(%d)", node.event.Command, node.event.Pid)
+	completeLine := front + line
+	if len(node.children) == 0 {
+		return completeLine + "\n"
+	}
+
+	completeLine += "-+\n"
+	// We add one here so the '+' and '|' of the children are aligned.
+	front += strings.Repeat(" ", len(line)+1)
+
+	for _, child := range node.children {
+		completeLine += treeToString(child, front)
+	}
+
+	return completeLine
+}
+
+func PrintTree(processes []*types.Event) error {
+	containers := make(map[string][]*types.Event, len(processes))
+	for _, process := range processes {
+		_, ok := containers[process.Container]
+		if !ok {
+			containers[process.Container] = make([]*types.Event, 0)
+		}
+		containers[process.Container] = append(containers[process.Container], process)
+	}
+
+	for _, container := range containers {
+		tree, err := createTree(container)
+
```

Comment:
```suggestion
```

Reply:
Why they are different semantically if the second line is checking the error that was just generated in the first one?